src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -122,14 +122,19 @@ const submitEncryptModeForm: SubmitEncryptModeFormHandlerType = async (
     ensureFormWithinSubmissionLimits,
   )
 
-  const hasEnsuredAll = await ensurePipeline.execute({
+  const ensuresError = await ensurePipeline.execute({
     form,
     logMeta,
     req,
     res,
   })
 
-  if (!hasEnsuredAll) return
+  if (ensuresError) {
+    const { errorMessage, statusCode } = mapRouteError(ensuresError)
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
```

Comment:
Ah, got it that it's not how pipelines usually behave. I think it's odd that we're adding code to the `if (!hasEnsuredAll)` branch even it's impossible to fulfil that condition.

Will it be better if we follow how Express middlewares handle their errors? What they do is to pass errors into the next function, then centralise error handling with an error handling middleware, as described [here](https://gist.github.com/zcaceres/2854ef613751563a3b506fabce4501fd).

But either way, I think this can be left for another PR bah! I'll revert my refactoring commit regarding ensures and add the catch-all as you suggested. We can make further changes in a separate PR if needed.

cc @KenLSM, I think you'll be interested in this discussion as well.

Reply:
Previous commit reverted in [09b367b](https://github.com/opengovsg/FormSG/pull/6595/commits/09b367b6f8a7544c82d921714b424f051a91a56b), suggested catch-all response return for errors implemented in [b9cf27a](https://github.com/opengovsg/FormSG/pull/6595/commits/b9cf27a62cd4186cc99f777b37e1d4e75fd04d4c).

Reply:
thanks @LinHuiqing , agree we can take this for another PR to keep this within scope